// @ui/eslint-config/tslint.js
module.exports = {
    overrides: [
        {
            files: ['*.ts', '*.tsx'],
            parser: '@typescript-eslint/parser',
            parserOptions: {
                jsx: true,
                useJSXTextNode: true,
                project: 'tsconfig.json',
                tsconfigRootDir: '.',
            },
            plugins: ['@typescript-eslint'],
            extends: [
                'airbnb',
                'airbnb/hooks',
                'plugin:@typescript-eslint/recommended',
                'plugin:@typescript-eslint/recommended-requiring-type-checking',
                'plugin:prettier/recommended',
            ],
            settings: {
                'import/resolver': {
                    node: {
                        extensions: ['.js', '.jsx', '.ts', '.tsx'],
                        moduleDirectory: ['node_modules', 'src/'],
                    },
                },
            },
            rules: {
                'max-len': [
                    'warn',
                    {
                        code: 120,
                        comments: 120,
                    },
                ],
                'no-trailing-spaces': 'error',
                '@typescript-eslint/explicit-function-return-type': 'off',
                '@typescript-eslint/explicit-module-boundary-types': 'off',
                "react/default-props-match-prop-types": "off",
                "react/function-component-definition": "off",
                'react/jsx-indent': ['error', 4],
                'import/extensions': 0,
                'import/prefer-default-export': 'off',
                'no-use-before-define': 'off',
                '@typescript-eslint/no-use-before-define': ['error'],
                'react/jsx-filename-extension': [
                    'off',
                    {
                        extensions: ['.tsx'],
                    },
                ],
                'react/require-default-props': "off",
                "no-unused-expressions": "off",
                'jsx-a11y/click-events-have-key-events': 'off',
                'jsx-a11y/no-static-element-interactions': 'off',
                'react/jsx-props-no-spreading': 'off',
                'no-unused-vars': 'off',
                '@typescript-eslint/no-unused-vars': ['error'],
            },
        },
    ],
};
